File: 3951-minimum-energy-to-maintain-brightness/3951-minimum-energy-to-maintain-brightness.java
class Solution {
    public long minEnergy(int n, int brightness, int[][] iv) {
        Arrays.sort(iv, (a, b) -> Integer.compare(a[0], b[0]));

        long cot = 0;

        int cs = iv[0][0];
        int ce = iv[0][1];

        for (int i = 1; i < iv.length; i++) {
            int start = iv[i][0];
            int end = iv[i][1];

            if (start <= ce + 1L) {
                ce = Math.max(ce, end);
            } else {
                cot += (long) ce - cs + 1;
                cs = start;
                ce = end;
            }
        }

        cot += (long) ce - cs + 1;

        long bulbsNeededPerTime = (brightness + 2L) / 3L;

        return cot * bulbsNeededPerTime;
    }
}

File: 3951-minimum-energy-to-maintain-brightness/3951-minimum-energy-to-maintain-brightness.py
class Solution:
    def minEnergy(self, n: int, B: int, T: list[list[int]]) -> int:
        res = 0
        r = -1
        for l, e in sorted(T):
            if l > r:
                res += e - l + 1
            elif e > r:
                res += e - r
            r = max(r, e)
        return res * ((B + 2) // 3)